### PR TITLE
Focus 254

### PR DESCRIPTION
**Pull request by rbadamsjr**
Created TaskTable, fixed Dashboard.js, TaskList, removed TaskWidget, fixed typo in Client code

### CODE REVIEW

File: frontend/src/pages/Dashboard.js [hunk 1]
```diff
@@ -1,10 +1,11 @@
 import { GoalsSnapshot } from "features/goals";
 import { ImpactsSnapshot } from "features/impacts";
 import { InitiativesSnapshot } from "features/initiatives";
-import { Widget } from "components";
+import { Widget, ClientInventory } from "components";
 import { NewsList } from "features/news/index";
-import { Login } from '@microsoft/mgt-react';
+import { Login } from "@microsoft/mgt-react";
 import { ScheduleList } from "features/schedule";
+import { TaskList, TasksWidget } from "features/tasks";
```
Comment (jsnkle) [inline frontend/src/pages/Dashboard.js:8]: I think we can remove TasksWidget here correct?

File: frontend/src/features/tasks/TaskTable.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
+import { useState } from "react";
+import { Link } from "react-router-dom";
+import {
+	useGetMyTasksQuery,
+	useUpdateTaskMutation,
```
Comment (jsnkle) [inline frontend/src/features/tasks/TaskTable.js:5]: doesn't look like useUpdateTaskMutation is being used

File: frontend/src/features/tasks/TaskTable.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+	};
+
+	let consultantId = 1; // Currently the auth data only stores the email address not the user id
+	const { data, isSuccess } = useGetMyTasksQuery(consultantId, {
+		refetchOnMountOrArgChange: true,
+	});
+
+	if (isSuccess) {
+		taskLists = data.tasklist;
+		taskTableColumns = Object.keys(taskLists[0]);
```
Comment (jsnkle) [inline frontend/src/features/tasks/TaskTable.js:47]: Where is taskTableColumns being used?

File: frontend/src/features/tasks/TaskTable.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+				<thead>
+					<tr>
+						<th>Title</th>
+						<th>Status</th>
+						<th>End Date</th>
+						<th></th>
+					</tr>
+				</thead>
+				<tbody>
+					{taskLists.map((row, idx) => {
```
Comment (jsnkle) [inline frontend/src/features/tasks/TaskTable.js:67]: what is idx being used for? not seeing that in here...

File: frontend/src/features/tasks/TaskTable.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+								)}
+								<td>{formatDate(row.end_date)}</td>
+								<td className='actions'>
+									<span className='action-item'>
+										<i
+											className='bi bi-pencil-square'
+											data-bs-toggle='tooltip'
+											data-bs-placment='top'
+											title='Edit'
+											type='button'
```
Comment (jsnkle) [inline frontend/src/features/tasks/TaskTable.js:86]: Not 100% sure we can use 'type' on an <i/> tag, can you check that please?

File: frontend/src/features/tasks/TaskTable.js [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+											onClick={() => editTaskHandle(row.id)}
+										></i>
+									</span>
+									<span className='action-item'>
+										<i
+											className='bi bi-trash-fill'
+											data-bs-placment='center'
+											title='Delete'
+											type='botton'
```
Comment (jsnkle) [inline frontend/src/features/tasks/TaskTable.js:95]: Same question as above on type attribute

File: frontend/src/features/tasks/TaskTable.js [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+	const [taskId, setTaskId] = useState();
+	const [model, setModel] = useState(false);
+
+	const [deleteTask] = useDeleteTaskMutation();
+
+	const deleteTaskHandle = async () => {
+		console.log("Delete task: " + taskId);
+		if (taskId) {
+			await deleteTask(taskId);
+			setTaskId();
```
Comment (jsnkle) [inline frontend/src/features/tasks/TaskTable.js:25]: I think you need an argument here... can't leave it empty but can set to null if you think it's necessary.

File: frontend/src/features/tasks/_tasks.scss [hunk 8]
```diff
@@ -15,3 +15,51 @@
 		margin-left: 5px;
 	}
 }
+
+table {
```
Comment (jsnkle) [inline frontend/src/features/tasks/_tasks.scss:19]: Was bootstrap not doing what you needed? Why the additional styling on table element?

File: frontend/src/pages/Dashboard.js [hunk 9]
```diff
@@ -1,10 +1,11 @@
 import { GoalsSnapshot } from "features/goals";
 import { ImpactsSnapshot } from "features/impacts";
 import { InitiativesSnapshot } from "features/initiatives";
-import { Widget } from "components";
+import { Widget, ClientInventory } from "components";
 import { NewsList } from "features/news/index";
-import { Login } from '@microsoft/mgt-react';
+import { Login } from "@microsoft/mgt-react";
```
Comment (jsnkle) [inline frontend/src/pages/Dashboard.js:6]: Please check your Prettier settings and confirm you have 'useTabs' set to true and 'jsxSingleQuote' set to true

**Review by jsnkle — CHANGES_REQUESTED** [review]
Linter is complaining about your arrow function on line 41 of the TaskList.js expecting a value to be returned. I think you can get away with an anonymous function there. Other comments in the files.

File: frontend/src/pages/Dashboard.js (hunk 1 above)
Comment (rbadamsjr) [reply frontend/src/pages/Dashboard.js:8]: Done

File: frontend/src/features/tasks/TaskTable.js (hunk 2 above)
Comment (rbadamsjr) [reply frontend/src/features/tasks/TaskTable.js:5]: Done

File: frontend/src/features/tasks/TaskTable.js (hunk 3 above)
Comment (rbadamsjr) [reply frontend/src/features/tasks/TaskTable.js:47]: Removed

File: frontend/src/features/tasks/TaskTable.js (hunk 4 above)
Comment (rbadamsjr) [reply frontend/src/features/tasks/TaskTable.js:67]: Originally used the idx to give the <td> a unique key.. but don't need it anymore

File: frontend/src/features/tasks/TaskTable.js (hunk 5 above)
Comment (rbadamsjr) [reply frontend/src/features/tasks/TaskTable.js:86]: No problem...

File: frontend/src/features/tasks/TaskTable.js (hunk 6 above)
Comment (rbadamsjr) [reply frontend/src/features/tasks/TaskTable.js:95]: removed

File: frontend/src/features/tasks/TaskTable.js (hunk 7 above)
Comment (rbadamsjr) [reply frontend/src/features/tasks/TaskTable.js:25]: Fixed ... also removed console.log()

File: frontend/src/features/tasks/_tasks.scss (hunk 8 above)
Comment (rbadamsjr) [reply frontend/src/features/tasks/_tasks.scss:19]: I wanted more space between the checkbox and the label

File: frontend/src/pages/Dashboard.js (hunk 9 above)
Comment (rbadamsjr) [reply frontend/src/pages/Dashboard.js:6]: Set it to true

File: frontend/src/features/tasks/_tasks.scss (hunk 8 above)
Comment (jsnkle) [reply frontend/src/features/tasks/_tasks.scss:19]: Sorry... I'm referring to the table element. You've added a width, layout, overflow and border attributes. Lines 19-23 and 36-40.

**Review by ArushiSharma95 — APPROVED** [review]
Looks good.

**Review by jsnkle — APPROVED** [review]

**Pull request merged**